lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -116,13 +121,21 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
 
       for (Operation &op : module.getBodyBlock()->getOperations())
         if (transferFunc(op).failed())
-          return signalPassFailure();
+          continue;
 
       // Record all the RefType ports to be removed later.
       for (size_t portNum = 0, e = module.getNumPorts(); portNum < e; ++portNum)
         if (module.getPortType(portNum).isa<RefType>())
           refPortsToRemoveMap[module].push_back(portNum);
     }
+    firstPass = false;
+    DenseSet<InstanceGraphNode *> visited;
+    for (auto *current : instanceGraph)
```

Comment:
This two-pass approach means a ref can go up and then down, but since the lattice tracks reaching send ops then a reftype flow that goes down and then up won't work, if I'm following (incl up, down, up)?

Is this an intentional limitation (doesn't seem unreasonable, certainly at first), or would we want to support more complicated flows through the circuit in the future?

Reply:
Right, I don't think we want to support u-turn kind of situation. 
But you are right, in case we need more complex flow, either this needs to run till convergence or there might be some simplification that can be done to track the ref.send.

Reply:
Sounds good!